src/nvme/tree.c

```diff
@@ -1014,16 +1012,8 @@ struct nvme_ctrl *nvme_create_ctrl(nvme_root_t r,
 		c->cfg.host_iface = strdup(host_iface);
 	if (trsvcid)
 		c->trsvcid = strdup(trsvcid);
-	else if (discovery)
-		discovery_trsvcid(c);
-	else if (!strncmp(transport, "rdma", 4) ||
-		 !strncmp(transport, "tcp", 3)) {
-		nvme_msg(r, LOG_ERR, "No trsvcid specified for '%s'\n",
-			 transport);
-		errno = EINVAL;
-		__nvme_free_ctrl(c);
-		c = NULL;
-	}
+	else
+		set_default_trsvcid(c);
```

Comment:
Sure this is correct?
Spec states the '4420' is the default port for RDMA, and 8009 is the default _discovery_ port for TCP.
So we cannot eliminate the check for discovery for tcp.

Reply:
What about the check in set_default_trsvcid() itself. Isn't this taking care of the discovery port for TCP being 8009 and IO ports on 4420?

Reply:
Isn't that already handled in set_default_trsvcid()? 
For NVMe/TCP - discovery controllers would use port 8009, whereas I/O controllers would use 4420
For NVMe/RDMA - both discovery controllers & I/O controllers would use 4420

Reply:
> Sure this is correct? Spec states the '4420' is the default port for RDMA, and 8009 is the default _discovery_ port for TCP. So we cannot eliminate the check for discovery for tcp.

[](https://github.com/martin-gpy)Isn't that already handled in set_default_trsvcid()?
For NVMe/TCP - discovery controllers would use port 8009, whereas I/O controllers would use 4420
For NVMe/RDMA - both discovery controllers & I/O controllers would use 4420
@martin-gpy

Reply:
Ah, right, you are correct.